### PR TITLE
Feature/profile questions 1

### PR DESCRIPTION
**Pull request by pavlo-chernykh**
- finished task 'UI page 1
image
- added responsive layout for laptop (1024px), tablet(768px), mobile(L,M,S)
image
image
-fixed constants import as you said before (now import 1 object containing all pages' constants)
image

### CODE REVIEW

File: apps/freelance/src/pages/ProfileQuestions1/index.tsx [hunk 1]
```diff
@@ -1,103 +1,157 @@
...
 
 const ProfileQuestions1 = () => {
   const { t } = useTranslation();
+  const { handleSubmit } = useForm<IProfileQuestions1>();
+  const [form] = Form.useForm();
+  const [AddprofileQuestions1Data] = useAddprofileQuestions1DataMutation();
+
   const { Option } = Select;
 
   const formItemLayout = {
```
Comment (lyudad) [inline apps/freelance/src/pages/ProfileQuestions1/index.tsx]: you will have constant initialisation with each rerender. its very very bad practice. all constants should be outside the component

File: apps/freelance/src/pages/ProfileQuestions1/index.tsx [hunk 2]
```diff
@@ -1,103 +1,157 @@
...
+          work_history_descr: workHistoryDescr,
+          work_history_from: workHistoryFrom,
+          work_history_to: workHistoryTo,
+        });
+        form.resetFields();
+      }
+    } catch (error) {
+      alert(error);
+    }
+  };
```
Comment (lyudad) [inline apps/freelance/src/pages/ProfileQuestions1/index.tsx:32]: you have too match logic - move to custom hook

File: apps/freelance/src/pages/ProfileQuestions1/index.tsx [hunk 3]
```diff
@@ -1,103 +1,157 @@
...
+        onFinish={values =>
+          handleSubmit(onFinish(values as IProfileQuestions1))
+        }
       >
         <Form.Item
           label={t('description.profileQp1.hR')}
-          name={profileQ1HR}
+          name={profileQ1.profileQ1HR}
           rules={[
             { required: true, message: 'Please, input your hourly rate in $!' },
```
Comment (lyudad) [inline apps/freelance/src/pages/ProfileQuestions1/index.tsx]: all texts should be in text file

File: apps/freelance/src/redux/profileQuestions/profileQuestions1Api.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+
+export const profileQuestions1Api = createApi({
+  reducerPath: 'profileQuestions1Api',
+  baseQuery: fetchBaseQuery({ baseUrl: 'http://localhost:5000' }),
```
Comment (lyudad) [inline apps/freelance/src/redux/profileQuestions/profileQuestions1Api.ts]: today you have localhos, tomorrow https://mysuperwebsite you will all the time change URL?
Please read what it env variables are, why we need to use it, and when. Create the list for me: in which case you will use env, in which constants, and when you will write something directly in the code, and send it to me in private message, without this work I will not accept your PR)

File: apps/freelance/src/pages/ProfileQuestions1/index.tsx (hunk 1 above)
Comment (pavlo-chernykh) [reply apps/freelance/src/pages/ProfileQuestions1/index.tsx]: fixed

File: apps/freelance/src/pages/ProfileQuestions1/index.tsx (hunk 2 above)
Comment (pavlo-chernykh) [reply apps/freelance/src/pages/ProfileQuestions1/index.tsx:32]: fixed (dunno if it's super correct to call it hook, though it has the same logic). Logic moved, code refactored a lot

File: apps/freelance/src/pages/ProfileQuestions1/index.tsx (hunk 3 above)
Comment (pavlo-chernykh) [reply apps/freelance/src/pages/ProfileQuestions1/index.tsx]: fixed

File: apps/freelance/src/redux/profileQuestions/profileQuestions1Api.ts (hunk 4 above)
Comment (pavlo-chernykh) [reply apps/freelance/src/redux/profileQuestions/profileQuestions1Api.ts]: fixed

File: apps/freelance/src/pages/ProfileQuestions1/hooks.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+import { IProfileQuestions1 } from './model';
+
+export const onFinishLogic = (values: IProfileQuestions1): profileQ1Payload => {
+  const jsonedEdu = JSON.parse(JSON.stringify(values.education));
+  const jsonedWork = JSON.parse(JSON.stringify(values.work_history_wrapper));
+  const returnedValues = {
+    ...values,
+    education_descr: jsonedEdu.information_about_education,
+    education_from: jsonedEdu.education_from.substring(0, 4),
```
Comment (lyudad) [inline apps/freelance/src/pages/ProfileQuestions1/hooks.ts]: it look like something very bad, why you need substring, try to find how do it without substring

File: apps/freelance/src/pages/ProfileQuestions1/hooks.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { profileQ1Payload } from 'redux/services/types/profileQuestions1.types';
+
+import { IProfileQuestions1 } from './model';
+
+export const onFinishLogic = (values: IProfileQuestions1): profileQ1Payload => {
+  const jsonedEdu = JSON.parse(JSON.stringify(values.education));
```
Comment (lyudad) [inline apps/freelance/src/pages/ProfileQuestions1/hooks.ts]: look weared, try to find solution without do difficult transformation

File: apps/freelance/src/pages/ProfileQuestions1/hooks.ts (hunk 5 above)
Comment (pavlo-chernykh) [reply apps/freelance/src/pages/ProfileQuestions1/hooks.ts]: fixed - added new interfaces for objects and it works (the problem was with typescript)

File: apps/freelance/src/pages/ProfileQuestions1/hooks.ts (hunk 6 above)
Comment (pavlo-chernykh) [reply apps/freelance/src/pages/ProfileQuestions1/hooks.ts]: fixed (added Moment library and fixed typescript)

File: apps/freelance/src/redux/profileQuestions/profileQuestions1Api.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
+import { profileQ1Payload } from 'redux/services/types/profileQuestions1.types';
+
+declare let process: {
+  env: {
+    NX_API_URL: string;
+  };
+};
+const baseUrl: string = (process.env.NX_API_URL as string) || '';
```
Comment (lyudad) [inline apps/freelance/src/redux/profileQuestions/profileQuestions1Api.ts]: move it to a constant file. in other case you will have a lot of same duplications

Comment (pavlo-chernykh) [reply apps/freelance/src/redux/profileQuestions/profileQuestions1Api.ts]: fixed

**Pull request merged**